vllm/v1/core/sched/scheduler.py

```diff
@@ -650,6 +646,17 @@ def update_from_output(
                 request.structured_output_request.grammar.accept_tokens(  # type: ignore[union-attr]
                     req_id, new_token_ids)
 
+            # Add newly generated spec token ids to the request.
+            if spec_token_ids is not None:
+                if request.use_structured_output:
+                    metadata = request.structured_output_request
+                    assert metadata is not None and metadata.grammar is not None
+                    # Needs to happen after new_token_ids are accepted.
+                    request.spec_token_ids = metadata.grammar.validate_tokens(
```

Comment:
When we trim the spec_token_ids, do we not need to udpate the `scheduled_spec_decode_tokens` [here](https://github.com/vllm-project/vllm/blob/54a66e5fee4a1ea62f1e4c79a078b20668e408c6/vllm/v1/core/sched/scheduler.py#L242-L251)?

Reply:
I think you may be confused. This code is in `update_from_output`, where the output of the model runner is received and processed. So, this is the first time the scheduler is seeing the proposed speculative tokens which are generated in the model runner. We are trimming them here for the first time. 

In the next iteration of the scheduler, the `request.spec_token_ids` will be the result of this trimmed value. Thus `scheduled_spec_decode_tokens` should have the right value when it is computed in `schedule(self)`.